src/EventSubscriber/ApqSubscriber.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Drupal\graphql\EventSubscriber;
+
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\graphql\Event\OperationEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use GraphQL\Error\Error;
+
+/**
+ * Save persisted queries to cache.
+ */
+class ApqSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The cache to store persisted queries.
+   *
+   * @var \Drupal\Core\Cache\CacheBackendInterface
+   */
+  protected $cache;
+
+  /**
+   * Constructs a ApqSubscriber object.
+   *
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cache
+   *   The cache to store persisted queries.
+   */
+  public function __construct(CacheBackendInterface $cache) {
+    $this->cache = $cache;
+  }
+
+  /**
+   * Handle operation start events.
+   *
+   * @param \Drupal\graphql\Event\OperationEvent $event
+   *   The kernel event object.
+   *
+   * @throws \GraphQL\Error\Error
+   */
+  public function onBeforeOperation(OperationEvent $event): void {
+    if (!array_key_exists('automatic_persisted_query', $event->getContext()->getServer()->getPersistedQueryInstances() ?? [])) {
+      return;
+    }
+    $query = $event->getContext()->getOperation()->query;
+    $queryHash = $event->getContext()->getOperation()->extensions['persistedQuery']['sha256Hash'] ?? '';
+
+    if ($query && $queryHash) {
```

Comment:
```suggestion
    if (is_string($query) && is_string($queryHash) && $queryHash !== '') {
```

This achieves the same and is slightly more explicit. I believe using falsey checks is discouraged by higher PHPStan levels.